### PR TITLE
Split TextInput into TextInputOutlined and TextInputFlat

### PR DESCRIPTION
**Pull request by pbitkowski**
What this PR consists of:
- make TextInput proxy component that renders TextInputOutlined or TextInputFlat based on 'mode' props,
- move TextInput components to its own directory and change imports in index to make it work.

### Motivation

Issue #791 

### Test plan

Open example and compare it with version without my changes.

### CODE REVIEW

**Comment by pbitkowski** [comment]
I have an issue with Flow here. It logs a long list of errors connected to possible usage of TextInput (from React Native) props on our TextInput. I used the same Properties as before so it's weird that it doesn't work. I'm not sure if the intersaction of `React.ElementConfig` and Object exact type is correct in TextInput props. Could you check it @ferrannp ? Thanks :)

**Comment by satya164** [comment]
I think the state can be moved up to the `TextInput` component to reduce duplication. Let's also move the types to a common file. The documentation can be removed from the individual components as well.

**Comment by pbitkowski** [comment]
@satya164 I moved types to common file and used spread operator for extending React Native TextInput Props but still got errors, which is weird. BTW. I rebased my branch to master and some TS errors appeared.

**Comment by pbitkowski** [comment]
Fixed flow errors and rebased. Could you take a look @Trancever ?

File: src/components/TextInput/TextInputOutlined.js [hunk 1]
```diff
@@ -501,24 +315,24 @@ class TextInput extends React.Component<Props, State> {
...
+              borderRadius: theme.roundness,
+              borderWidth: hasActiveOutline ? 2 : 1,
+              borderColor: hasActiveOutline ? activeColor : outlineColor,
+            },
+          ]}
+        />
 
-        {mode === 'outlined' && label ? (
+        {label ? (
           // When mode == 'outlined', the input label stays on top of the outline
```
Comment (Trancever) [inline src/components/TextInput/TextInputOutlined.js]: We can remove `When mode == 'outlined` from comment :D

Comment (pbitkowski) [reply src/components/TextInput/TextInputOutlined.js]: Done

File: src/components/TextInput/TextInputFlat.js [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+              0,
+            ],
+          }),
+        },
+      ],
+    };
+
+    return (
+      <View style={[containerStyle, style]}>
+        {/* When mode === 'flat', render an underline */}
```
[the remark this answers is not in the document]
Comment (pbitkowski) [reply src/components/TextInput/TextInputFlat.js]: Done

File: src/components/TextInput/TextInput.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+ */
+
+class TextInput extends React.Component<TextInputProps, State> {
+  static defaultProps = {
+    mode: 'flat',
+    disabled: false,
+    error: false,
+    multiline: false,
+    editable: true,
+    render: props => <NativeTextInput {...props} />,
```
[the remark this answers is not in the document]
Comment (pbitkowski) [reply src/components/TextInput/TextInput.js]: Done

**Pull request merged**